src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1

```diff
@@ -317,55 +317,98 @@ class AgentPool: ADOSVTBase
                             }
-                            else 
+                            else
                             {
                                 $this.agentPoolActivityDetail.isAgentPoolActive = $true;
                                 $this.agentPoolActivityDetail.message = "Agent pool has been queued in the last $inactiveLimit days.";
                             }
                             $this.agentPoolActivityDetail.agentPoolLastRunDate = $agtPoolLastRunDate;
                         }
-                        else 
+                        else
                         {
                             $this.agentPoolActivityDetail.isAgentPoolActive = $true;
                             $this.agentPoolActivityDetail.message = "Agent pool was being queued during control evaluation.";
                         }
                 }
-                else 
+                else
                 {
                     #[else] Agent pool is created but nenver run, check creation date greated then 180
                     $this.agentPoolActivityDetail.isAgentPoolActive = $false;
-                    if (([Helpers]::CheckMember($this.agentPool, "fps.dataProviders.data") ) -and ($this.agentPool.fps.dataProviders.data."ms.vss-build-web.agent-pool-data-provider")) 
+                    if (([Helpers]::CheckMember($this.agentPool, "fps.dataProviders.data") ) -and ($this.agentPool.fps.dataProviders.data."ms.vss-build-web.agent-pool-data-provider"))
                     {
                         $agentPoolDetails = $this.agentPool.fps.dataProviders.data."ms.vss-build-web.agent-pool-data-provider"
                         $this.agentPoolActivityDetail.agentPoolCreationDate = $agentPoolDetails.selectedAgentPool.createdOn;
                     }
-                    else 
+                    else
                     {
                         $this.agentPoolActivityDetail.message = "Could not fetch agent pool details.";
-                    }                    
+                    }
                 }
             }
-            else 
+            else
             {
                 $this.agentPoolActivityDetail.message = "Could not fetch agent pool details.";
             }
         }
-        catch 
+        catch
         {
             $this.agentPoolActivityDetail.message = "Could not fetch agent pool details.";
         }
         $this.agentPoolActivityDetail.isComputed = $true
     }
+
+    hidden [ControlResult] CheckBroaderGroupAccess ([ControlResult] $controlResult) {
+        $failMsg = $null
+        try {
+            $restrictedGroups = @();
+
+            if ($this.ControlSettings -and [Helpers]::CheckMember($this.ControlSettings, "AgentPool.RestrictedBroaderGroupsForAgentpool") ) {
+                $restrictedGlobalGroupsForSerConn = $this.ControlSettings.AgentPool.RestrictedBroaderGroupsForAgentpool;
```

Comment:
rename this 'restrictedGlobalGroupsForSerConn' variable itz for agentpools

Reply:
Addressed